specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json

```diff
@@ -0,0 +1,257 @@
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DefenderForStorageSettingProperties"
+        }
+      },
+      "description": "The Defender for Storage resource."
+    },
+    "DefenderForStorageSettingProperties": {
+      "type": "object",
+      "properties": {
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Indicates whether Defender for Storage is enabled on this resource."
+        },
+        "malwareScanning": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/MalwareScanningProperties"
+        },
+        "sensitiveDataDiscovery": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SensitiveDataDiscoveryProperties"
+        },
+        "overrideSubscriptionLevelSettings": {
+          "type": "boolean",
+          "description": "Indicates whether the settings defined for this resource should override the settings defined on the subscription."
+        }
+      },
+      "description": "A Defender for Storage resource properties."
+    },
+    "MalwareScanningProperties": {
+      "type": "object",
+      "properties": {
+        "onUpload": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/OnUploadProperties"
+        },
+        "operationStatus": {
+          "type": "object",
+          "$ref": "#/definitions/OperationStatus",
+          "readOnly": true,
+          "description": "Optional. A status describing the success/failure of Malware Scanning enablement/disablement operation."
+        }
+      },
+      "description": "Properties of Malware Scanning."
+    },
+    "OnUploadProperties": {
+      "type": "object",
+      "properties": {
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Indicates whether On Upload malware scanning should be enabled."
+        },
+        "capGBPerMonth": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Defines the max GB to be scanned per Month. Set to -1 if no capping is needed."
+        }
+      },
+      "description": "Properties of On Upload malware scanning."
+    },
+    "SensitiveDataDiscoveryProperties": {
+      "type": "object",
+      "properties": {
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Indicates whether Sensitive Data Discovery should be enabled."
+        },
+        "operationStatus": {
+          "type": "object",
+          "$ref": "#/definitions/OperationStatus",
+          "readOnly": true,
+          "description": "Optional. A status describing the success/failure of Sensitive Data Discovery's enablement/disablement operation."
+        }
+      },
+      "description": "Properties of Sensitive Data Discovery."
+    },
+    "OperationStatus": {
```

Comment:
Why is the capitalization of this OperationStatus different from the others?

Reply:
The convention is that definitions are capitalized, but the properties inside the definitions are not. This is what I did in this file.
This is relevant for your other comments as well...